src/statistics.rs

```diff
@@ -0,0 +1,771 @@
+    let config_exists = context
+        .sql
+        .get_raw_config(Config::StatsLastOldContactId.as_ref())
+        .await?
+        .is_some();
+    if config_exists {
+        // The user had statistics-sending enabled already in the past,
+        // keep the 'last old contact id' as-is
+        return Ok(());
+    }
+
+    let last_contact_id: u64 = context
+        .sql
+        .query_get_value("SELECT MAX(id) FROM contacts", ())
+        .await?
+        .unwrap_or(0);
+
+    context
+        .sql
+        .set_raw_config(
+            Config::StatsLastOldContactId.as_ref(),
+            Some(&last_contact_id.to_string()),
+        )
+        .await?;
+
+    Ok(())
+}
+
+async fn get_statistics(context: &Context) -> Result<String> {
+    // The ID of the last msg that was already counted in the previously sent statistics.
+    // Only newer messages will be counted in the current statistics.
+    let last_counted_msg = context
+        .get_config_u32(Config::StatsLastCountedMsgId)
+        .await?;
+
+    // The Id of the last contact that already existed when the user enabled the setting.
+    // Newer contacts will get the `new` flag set.
+    let last_old_contact = context
+        .get_config_u32(Config::StatsLastOldContactId)
+        .await?;
+
+    let key_create_timestamps: Vec<i64> = load_self_public_keyring(context)
+        .await?
+        .iter()
+        .map(|k| k.created_at().timestamp())
+        .collect();
+
+    let statistics_id = match context.get_config(Config::StatsId).await? {
+        Some(id) => id,
+        None => {
+            let id = create_id();
+            context
+                .set_config_internal(Config::StatsId, Some(&id))
+                .await?;
+            id
+        }
+    };
+
+    let statistics = Statistics {
+        core_version: get_version_str().to_string(),
+        key_create_timestamps,
+        statistics_id,
+        is_chatmail: context.is_chatmail().await?,
+        contact_stats: get_contact_stats(context, last_old_contact).await?,
+        message_stats_one_one: get_message_stats(context, last_counted_msg, true).await?,
+        message_stats_multi_user: get_message_stats(context, last_counted_msg, false).await?,
+        securejoin_sources: get_securejoin_source_stats(context).await?,
+        securejoin_uipaths: get_securejoin_uipath_stats(context).await?,
+        securejoin_invites: get_securejoin_invite_stats(context).await?,
+    };
+
+    Ok(serde_json::to_string_pretty(&statistics)?)
+}
+
+async fn get_statistics_chat_id(context: &Context) -> Result<ChatId, anyhow::Error> {
+    let contact_id: ContactId = *import_vcard(context, STATISTICS_BOT_VCARD)
+        .await?
+        .first()
+        .context("Statistics bot vCard does not contain a contact")?;
+    mark_contact_id_as_verified(context, contact_id, ContactId::SELF).await?;
+
+    let chat_id = if let Some(res) = ChatId::lookup_by_contact(context, contact_id).await? {
```

Comment:
So if the bot key changes, we'll send statistics to another chat. Just for the record. Not sure that e.g. creating a broadcast/group and adding the bot there is better.